Clarify using table name in Security JPA guide (#46581)
quarkusio/quarkus: docs/src/main/asciidoc/security-jpa.adoc

The `quarkus-security-jpa` extension initializes only if a single entity is annotated with `@UserDefinition`.

> <1> In most databases, `user` is a restricted keyword, so define the table name as something else to avoid conflicts

Reviewer: ```suggestion
<1> Table name. Do not define it as `user`, which is a restricted keyword in most databases.
```

Minor suggestion as the callout should introduce what is defined here first

<2> The `@UserDefinition` annotation must be present on a single entity, either a regular Hibernate ORM entity or a Hibernate ORM with Panache entity.
<3> Indicates the field used for the username.
<4> Indicates the field used for the password.

> By default, `quarkus-security-jpa` uses bcrypt-hashed passwords, or you can configure plain text or custom passwords instead.

Reviewer: IMHO this note is not needed here, there is a dedicated section, https://quarkus.io/guides/security-jpa#password-storage-and-hashing.
Perhaps, updating the current line to `Indicates the field used for the password. See the <<password-hashing-and-storage>> for more information` can work ? (`password-hashing-and-storage` anchor should be present in that section or refer to the existing one if it is already available)
